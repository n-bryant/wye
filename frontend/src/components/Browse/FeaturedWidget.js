import React from "react";
import PropTypes from "prop-types";

import createClassNameHelper from "@n_bryant/classnames-helper";
import JSS_CLASS_NAME_PREFIX from "../../../lib/classNamePrefix";
import { withStyles } from "@material-ui/core/styles";

import Link from "next/link";

import Container from "@material-ui/core/Container";
import Typography from "@material-ui/core/Typography";
import Button from "@material-ui/core/Button";

import styles from "./FeaturedWidget.styles";

/**
 * renders a featured widget container with title, subtitle, action button and any content passed to it
 */
export const FeaturedWidget = props => {
  const classnames = FeaturedWidget.classnames(props);
  const { title, subTitle, seeMorePath } = props;

  return (
    <Container
      className={classnames.element("root")}
      disableGutters={true}
      maxWidth={false}
    >
      <Typography
        className={classnames.element("heading")}
        variant="h2"
        gutterBottom={true}
      >
        {title}
      </Typography>
      <Typography
        className={classnames.element("subHeading")}
        variant="body1"
        gutterBottom={true}
      >
        {subTitle}
      </Typography>
      <Container
        className={classnames.element("content")}
        disableGutters={true}
      >
        {props.children}
      </Container>
      {seeMorePath && (
        <Container
          className={classnames.element("seeMoreButtonContainer")}
          disableGutters={true}
        >
          <Link href={seeMorePath}>
            <a>
              <Button>
                <Typography variant="body2">See more...</Typography>
              </Button>
            </a>
          </Link>
        </Container>
      )}
    </Container>
  );
};
FeaturedWidget.classnames = createClassNameHelper(
  `${JSS_CLASS_NAME_PREFIX}FeaturedWidget`
);
FeaturedWidget.propTypes = {
  // styles to apply
  classes: PropTypes.shape({
    root: PropTypes.string,
    heading: PropTypes.string,
    subHeading: PropTypes.string,
    content: PropTypes.string,
    seeMoreButtonContainer: PropTypes.string
  }),
  // the title of the widget
  title: PropTypes.string.isRequired,
  // the sub-heading for the widget
  subTitle: PropTypes.string.isRequired,
  // link to a filtered browse page based on type of featured category
  seeMorePath: PropTypes.string
};
FeaturedWidget.defaultProps = {
  classes: {}
};

export default withStyles(styles)(FeaturedWidget);
