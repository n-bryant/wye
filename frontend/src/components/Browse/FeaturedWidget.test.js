import { FeaturedWidget } from "./FeaturedWidget";

describe("FeaturedWidget", () => {
  const props = {
    classes: {
      root: "root",
      heading: "heading",
      subHeading: "subHeading",
      content: "content",
      seeMoreButtonContainer: "seeMoreButtonContainer"
    },
    title: "title",
    subTitle: "subTitle",
    seeMorePath: "path"
  };

  it("should render successfully", () => {
    expect(toJson(shallow(<FeaturedWidget {...props} />))).toMatchSnapshot();
  });

  it("should not render the see more button if seeMorePath is falsy", () => {
    let propsWithoutSeeMorePath = {
      ...props
    };
    delete propsWithoutSeeMorePath.seeMorePath;
    const wrapper = shallow(<FeaturedWidget {...propsWithoutSeeMorePath} />);
    expect(
      wrapper.findWhere(n => n.hasClass(props.seeMoreButtonContainer)).length
    ).toBe(0);
  });
});
