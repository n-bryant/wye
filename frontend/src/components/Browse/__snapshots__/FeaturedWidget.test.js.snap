// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturedWidget should render successfully 1`] = `
<WithStyles(ForwardRef(Container))
  className="wye-FeaturedWidget__root root"
  disableGutters={true}
  maxWidth={false}
>
  <WithStyles(ForwardRef(Typography))
    className="wye-FeaturedWidget__heading heading"
    gutterBottom={true}
    variant="h2"
  >
    title
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(Typography))
    className="wye-FeaturedWidget__subHeading subHeading"
    gutterBottom={true}
    variant="body1"
  >
    subTitle
  </WithStyles(ForwardRef(Typography))>
  <WithStyles(ForwardRef(Container))
    className="wye-FeaturedWidget__content content"
    disableGutters={true}
  />
  <WithStyles(ForwardRef(Container))
    className="wye-FeaturedWidget__seeMoreButtonContainer seeMoreButtonContainer"
    disableGutters={true}
  >
    <Link
      href="path"
    >
      <a>
        <WithStyles(ForwardRef(Button))>
          <WithStyles(ForwardRef(Typography))
            variant="body2"
          >
            See more...
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(Button))>
      </a>
    </Link>
  </WithStyles(ForwardRef(Container))>
</WithStyles(ForwardRef(Container))>
`;
